### PR TITLE
Add totalCollateral parameter and getOptionInfo for short puts

### PR DESCRIPTION
**Pull request by zpv**
Adds `totalCollateral` field to ShortPut curves. This allows for contracts that are slightly unbacked (for fee / compounding purposes)

Add simple logic to differentiate between call / put curves. (If ascending, put, call otherwise)
This is not sound by itself. After extracting option info, we recreate the curve and check they are equivalent.

---

Closes ATOM-216
Closes ATOM-218
Closes ATOM-215

### CODE REVIEW

File: packages/core/lib/dlc/finance/OptionInfo.ts [hunk 1]
```diff
@@ -65,10 +71,26 @@ export function getOptionInfoFromContractInfo(
...
+  const curve = HyperbolaPayoutCurve.fromPayoutCurvePiece(payoutCurvePiece);
+  const maxOutcome = BigInt(
+    new BN(eventDescriptor.base)
+      .pow(eventDescriptor.nbDigits)
+      .minus(1)
+      .toString(10),
+  );
+  const isAscending = curve
+    .getPayout(maxOutcome)
+    .gt(Number(payoutFunction.endpointPayout0));
```
Comment (matthewjablack) [inline packages/core/lib/dlc/finance/OptionInfo.ts:83]: So for now it only checks if it's ascending or not to determine if it's a put or call? I suspect there are lots of edge cases where this is not the case. Any reason for not running it through `buildPayoutFunction` again?

Comment (zpv) [reply packages/core/lib/dlc/finance/OptionInfo.ts:83]: More secure differentiation is a separate task in Linear: https://linear.app/atomicfinance/issue/ATOM-216/implement-a-more-secure-way-of-differentiating-between-a-put-and-call

Currently in progress is rebuilding the curve and implementing and performing an `isEquals` check on the two curves.

Comment (zpv) [reply packages/core/lib/dlc/finance/OptionInfo.ts:83]: Added sanity check

File: packages/core/lib/dlc/finance/ShortPut.ts [hunk 2]
```diff
@@ -29,20 +30,22 @@ const buildCurve = (
...
-    totalCollateral: contractSize,
+    totalCollateral,
     payoutCurve,
   };
 };
 
 const buildPayoutFunction = (
   strikePrice: bigint,
   contractSize: bigint,
+  totalCollateral: bigint,
```
[the remark this answers is not in the document]
Comment (matthewjablack) [reply packages/core/lib/dlc/finance/ShortPut.ts:41]: > Is that preferable?
Ah, I see the issue now. Nope, I don't think that is preferable actually. Let's stick with different interfaces and we can decide to create a standard one down the line.

**Review by matthewjablack — APPROVED** [review]
LGTM 🎉

**Pull request merged**